### PR TITLE
Store `Config` keywords as map

### PR DESCRIPTION
**Pull request by rgmz**
### Description:
This uses a `map` to deduplicate `Config.Keywords`.

### Checklist:

* [x] Does your PR pass tests?
* [ ] Have you written new tests for your changes?
* [ ] Have you lint your code locally prior to submission?

### CODE REVIEW

**Comment by rgmz** [comment]
Hmm, lack of ordered output for the generated config could be more effort than it's worth.

**Comment by zricethezav** [comment]
> Hmm, lack of ordered output for the generated config could be more effort than it's worth.
 
 Is this still a concern?

**Comment by rgmz** [comment]
No. The original code only used a map in validate.go which resulted in random keyword order.

File: cmd/generate/config/utils/validate.go [hunk 1]
```diff
@@ -81,5 +60,29 @@ func ValidateWithPaths(r config.Rule, truePositives map[string]string, falsePosi
 				Msg("Failed to Validate. False positive was detected by regex and/or path.")
 		}
 	}
-	return &r
+	return r
+}
+
+func createDetector(r *config.Rule) *detect.Detector {
```
Comment (zricethezav) [inline cmd/generate/config/utils/validate.go:66]: **Suggestion**: @rgmz wdyt about renaming this `createSingleRuleDetector`? At first when I read this code I was like "wait why are we building a detector here?"

**Pull request merged**